addons/hr_expense/models/hr_expense.py

```diff
@@ -116,13 +116,14 @@ def attach_document(self, **kwargs):
 
     @api.depends('product_has_cost')
     def _compute_currency_id(self):
```

Comment:
I wasn't even aware this could work. It's a stored computed field depending on a non-stored computed field O.O

I don't know how it is in master now, but this clearly could and should be improved there ; having something being recomputed on all expenses using a product when this product is modified feels really bad technically speaking (even with your if in the compute: the function needs to be called on a potentially HUGE number of records) :/ The more expenses you'll have, the more things you'll have to remcopute. 

Not sure to understand why it's interesting to recompute the amount of draft expenses when the product's cost is changed ... Shouldn't it be fixed ? As soon as I made an expense, even in draft, if I set a field, I mean it to stay :thinking: We don't do that on draft invoices when a product price is changed, for example, so ... why here ? (this would probably need too many changes for stable ; just challenging the feature with master in mind)

Reply:
Agreed, we added a warning on the product recently recommending not to do that
I am fighting for the acceptance of more multi-currencies flows in expense but for now that is a PO request
There is indeed room for improvement there, I agree (same state in master iirc)

Reply:
@Laura-labe We'll need to discuss this "PO request" ^^